app/cruds/cruds_raffle.py

```diff
@@ -0,0 +1,429 @@
+    db: AsyncSession,
+) -> models_raffle.TypeTicket:
+    """Create a new Typeticket in databasend return it"""
+
+    db.add(typeticket)
+    try:
+        await db.commit()
+        return typeticket
+    except IntegrityError as err:
+        await db.rollback()
+        raise err
+
+
+async def get_typeticket_by_raffleid(
+    raffle_id: str,
+    db: AsyncSession,
+) -> list[models_raffle.TypeTicket] | None:
+    result = await db.execute(
+        select(models_raffle.TypeTicket).where(
+            models_raffle.TypeTicket.raffle_id == raffle_id
+        )
+    )
+    return result.scalars().all()
+
+
+async def get_typeticket_by_id(
+    typeticket_id: str,
+    db: AsyncSession,
+) -> models_raffle.TypeTicket | None:
+    result = await db.execute(
+        select(models_raffle.TypeTicket)
+        .where(models_raffle.TypeTicket.id == typeticket_id)
+        .options(selectinload(models_raffle.TypeTicket.raffle))
+    )
+    return result.scalars().first()
+
+
+async def edit_typeticket(
+    typeticket_id: str,
+    typeticket_update: schemas_raffle.TypeTicketEdit,
+    db: AsyncSession,
+):
+    await db.execute(
+        update(models_raffle.TypeTicket)
+        .where(models_raffle.TypeTicket.id == typeticket_id)
+        .values(**typeticket_update.dict(exclude_none=True))
+    )
+    await db.commit()
+
+
+async def delete_typeticket(
+    db: AsyncSession,
+    typeticket_id: str,
+):
+    """Delete a raffle from database by id"""
+
+    await db.execute(
+        delete(models_raffle.TypeTicket).where(
+            models_raffle.TypeTicket.id == typeticket_id
+        )
+    )
+    await db.commit()
+
+
+async def get_tickets(db: AsyncSession) -> list[models_raffle.Tickets]:
+    """Return all tickets from database where raffle ID == raffle_id"""
+
+    result = await db.execute(select(models_raffle.Tickets))
+    return result.scalars().all()
+
+
+async def create_ticket(
+    tickets: list[models_raffle.Tickets],
+    db: AsyncSession,
+) -> list[models_raffle.Tickets]:
+    """Create a new ticket in databasend return it"""
+    db.add_all(tickets)
+    try:
+        await db.commit()
+        return tickets
+    except IntegrityError as err:
+        await db.rollback()
+        raise err
+
+
+async def get_ticket_by_raffleid(
+    raffle_id: str,
+    db: AsyncSession,
+) -> list[models_raffle.Tickets]:
+    results = (
+        (
+            await db.execute(
+                select(models_raffle.Tickets).options(
+                    joinedload(models_raffle.Tickets.type_ticket).joinedload(
+                        models_raffle.TypeTicket.raffle
+                    ),
+                    joinedload(models_raffle.Tickets.user),
+                    joinedload(models_raffle.Tickets.lot),
+                )
+            )
+        )
+        .scalars()
+        .all()
+    )
+    filtered_results = [
+        result for result in results if result.type_ticket.raffle_id == raffle_id
```

Comment:
Is there a reason for not filtering with SQL alchemy ?

Reply:
We can fix this by adding the column at the same time we will rename tables (later for compatibility with frontend)